enterprise_gateway/services/processproxies/yarn.py

```diff
@@ -469,3 +493,75 @@ def _kill_app_by_id(self, app_id):
                              format(app_id, e))
 
         return response
+
+    def _check_queue_partition_usage(self, queue_name, partition_name, threshold):
+
+        candidate_queue = self.resource_mgr.cluster_scheduler_queue(queue_name)
+
+        if candidate_queue is None:
+            self.log.warning("Queue: {} not found in cluster.".format(queue_name))
+            return None
+
+        candidate_partition = self.resource_mgr.cluster_queue_partition(candidate_queue, partition_name)
+
+        if candidate_partition is None:
+            self.log.warning("Partition: {} not found in {} queue.".format(partition_name, queue_name))
+            return None
+
+        self.log.debug("Checking endpoint: {} if queue {} has used capacity <= {}% for the partition: {} "
+                       .format(self.resource_mgr.get_active_endpoint(), queue_name, threshold, partition_name))
+
+        queue_available = self.resource_mgr.cluster_scheduler_queue_availability(candidate_partition, threshold)
+        return queue_available
+
+    def _check_resource(self, gpu, memory_mb, cores, node_label):
+
+        response = self.resource_mgr.cluster_nodes(states=["RUNNING"])
+        content = response.data
+
+        available_resources = []
+        for node in content["nodes"]["node"]:
+            if node_label in node["nodeLabels"]:
+                resource = {}
+                for resource_info in node["availableResource"]["resourceInformations"]["resourceInformation"]:
+                    if resource_info['name'] == "yarn.io/gpu":
+                        resource['gpu'] = int(resource_info['value'])
+                    elif resource_info['name'] == "memory-mb":
+                        resource['memory-mb'] = int(resource_info['value'])
+                    elif resource_info['name'] == "vcores":
+                        resource['vcores'] = int(resource_info['value'])
+
+                available_resources.append(resource)
+
+        if len(available_resources) == 0:
+            self.log.warning("No running node found with the given partition name/ node label: {}.".format(node_label))
+            return None, None
+
+        is_available = False
+        is_gpu = False
+        is_memory_mb = False
+        is_cores = False
+
+        for resource in available_resources:
+            if resource['gpu'] >= gpu:
+                if resource['memory-mb'] >= memory_mb:
+                    if resource['vcores'] >= cores:
+                        is_available = True
+                        break
+                    else:
+                        is_cores = True
+                else:
+                    is_memory_mb = True
+            else:
+                is_gpu = True
+
+        if not is_available:
+            if is_cores:
+                return False, "cores"
```

Comment:
If we return `cores`, the error message would look like `The following Yarn compute resource is unavailable: cores after X secs`
This would be difficult to interpret for the user
We can either pass `driver_cpu`, which is the key in the hardware resource spec of hunch CLI, or just `Driver CPU cores`
(Same comment for `memory-mb` resource)

Reply:
Sure. I have changed the message.